lib/Unix/PID/Tiny.pm

```diff
@@ -242,9 +333,34 @@ Like Unix::PID but supplies only a few key functions.
 
 =head1 INTERFACE
 
-=head2 new()
+=head2 new(I<[$args_hr]>)
+
+See L<Unix::PID>'s new().  The following options can be provided in the
+optional HASH, I<$args_hr>, to enable certain extensions:
+
+=over
+
+=item B<keep_open>
+
+When a true value is provided, PID files created with pid_file() will remain
+open.
 
-See L<Unix::PID>'s new()
+=item B<check_proc_open_fds>
+
+When a true value is provided, this option will cause is_pidfile_running() to
+traverse C</proc/$pid/fd> to ensure a current file descriptor is held by a
+given process for the specified PID file.
+
+This option is only useful when B<keep_open> is also enabled.
```

Comment:
`check_proc_open_fds` seems to do its thing regardless of `keep_open` IILAIC.

Can this  be removed or should the `check_proc_open_fds` code factor in `keep_open`?

Reply:
I left the operation of check_proc_open_fds independent from keep_open because the pid file may possibly be supplied by something other than Unix::PID::Tiny itself.

Reply:
What does `keep_open` do on its own?

Reply:
The intent is to simply cause the file handle to remain open. close_on_destroy, used in conjunction with keep_open, will close the file handle when the blessed object is destroyed.

Reply:
I see, it just doesn’t close them explicitly.

Reply:
oh wait but the FH lexical so it will auto close at the `return 1;`. So I am puzzled as to the point of `keep_open` and `close_on_destroy` since they are essentially no-ops because of the auto-close-fh when they go out of scope behavior of perl. Maybe I am misunderstanding something?

Reply:
I was not aware that Perl lexical file handles had magic bound to their SV entry to cause them to close; honestly I've always closed file handles explicitly, no matter what, and never expected Perl to actually have this sort of behavior. I honestly believed that file descriptors would leak otherwise. I guess that's what I get for spending so much time writing C?

Reply:
yep ;)

```
{   open my $fh, "…", $whatever or …; doe anythign but close $fh here; }
handle is closed
```

Reply:
Thank you. I know. I tested it before I opened my dumb mouth.